### PR TITLE
feat: implemented hilt in the project

### PR DESCRIPTION
**Pull request by PratyushSingh07**
Jira Ticket : https://mifosforge.jira.com/browse/MOBILE-83

we can get rid of the `PerActivity` annotation and also the `ReviewLoanApplicationViewModelFactory` because now hilt will do that for us

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
hey @jawidMuhammadi could you please review this!!

File: app/src/main/java/org/mifos/mobile/ui/activities/HomeActivity.kt [hunk 1]
```diff
@@ -78,7 +80,7 @@ class HomeActivity :
     public override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityHomeBinding.inflate(layoutInflater)
-        activityComponent?.inject(this)
+        //activityComponent?.inject(this)
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/activities/HomeActivity.kt]: If this line of code is not required anymore then feel free to remove it.

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/activities/HomeActivity.kt (hunk 1 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/activities/HomeActivity.kt:83]: yes indeed , i must have missed it

Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/activities/HomeActivity.kt]: done

**Review by jawidMuhammadi — APPROVED** [review]
Looks good to me! Great job Partyush! 👏

**Pull request merged**